### PR TITLE
Release azure-cosmo:4.33.0

### PR DESCRIPTION
**Pull request by xinlian12**
* azure-cosmos 4.33.0 release
* azure-spring-data-cosmos 3.24.0 release
* azure-cosmos-spark: 4.12.0 relese
* azure-cosmos-encryption: 1.4.0 release
* Added few extra log lines to help debug a spark issue

### CODE REVIEW

**Review by FabianMeiswinkel — APPROVED** [review]
LGTM

**Review by kirankumarkolli — APPROVED** [review]

**Review by kushagraThapar — APPROVED** [review]

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/feedranges/FeedRangeEpkImpl.java [hunk 1]
```diff
@@ -175,10 +176,15 @@ public Mono<RxDocumentServiceRequest> populateFeedRangeFilteringHeaders(
                         }
 
                         final List<PartitionKeyRange> pkRanges = pkRangeHolder.v;
-                        if (pkRanges == null || pkRanges.size() == 0) {
+                        if (pkRanges == null) {
                             return Mono.error(new NotFoundException(
-                                String.format("Stale cache for collection rid '%s'.",
-                                    containerRid)));
+                                String.format("Stale cache for collection rid '%s': pkRanges are null",
```
Comment (FabianMeiswinkel) [inline sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/feedranges/FeedRangeEpkImpl.java:181]: In both cases I think we should also log this.range?

**Review by FabianMeiswinkel — APPROVED** [review]

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/feedranges/FeedRangeEpkImpl.java (hunk 1 above)
Comment (xinlian12) [reply sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/feedranges/FeedRangeEpkImpl.java]: added

**Comment by xinlian12** [comment]
The failed tests are due to emulator has starting issue

**Pull request merged**